vizro-ai/examples/dashboard_ui/components.py

```diff
@@ -386,16 +391,19 @@ def build(self):
             toggleClassName="dropdown-menu-toggle-class",
         )
         download_div = html.Div(
-            children=[
-                html.Span("download", className="material-symbols-outlined", id=f"{self.id}-icon"),
-                dropdown_menu,
-                dbc.Tooltip(
-                    "Download this plot to your device as a plotly JSON or interactive HTML file "
-                    "for easy sharing or future use.",
-                    target="dropdown-menu-icon",
-                ),
-            ],
-            id="dropdown-menu-id",
+            html.Div(
+                children=[
+                    html.Span("download", className="material-symbols-outlined", id=f"{self.id}-icon"),
+                    dropdown_menu,
+                    dbc.Tooltip(
+                        "Download this plot to your device as a plotly JSON or interactive HTML file "
+                        "for easy sharing or future use.",
+                        target="dropdown-menu-icon",
+                    ),
+                ],
+                id="dropdown-menu-div",
+            ),
+            className="dropdown-menu-outer-div",
```

Comment:
The download icon and the download label should be in the same container and the tooltip should be around the entire button - this way this looks like one component. 

Currently you get the tooltip if you hover of the icon only, but normally I would hover over the entire button instead of individual elements within. 


So something like this. You still have to customise some CSS for it to look good, but structurally the button will be one component, so the tooltip will appear when you hover over the entire button, not only the icon and the label consists of the icon and the text label. I am not sure if you need all of these html.divs - I've removed some of them below. This way it's easier to customise CSS and see which of these containers we really need

```
class DropdownMenu(vm.VizroBaseModel):
    """Custom dropdown menu component."""

    type: Literal["dropdown_menu"] = "dropdown_menu"

    def build(self):
        """Returns custom dropdown menu."""
        dropdown_menu = dbc.DropdownMenu(
            id="dropdown-menu-button",
            label=[html.Span("download", className="material-symbols-outlined", id=f"{self.id}-icon"), html.Span("Download")],
            children=[
                dbc.DropdownMenuItem(
                    children=[
                        dbc.Button(children="JSON", id=f"{self.id}-json", className="download-button"),
                    ]
                ),
                dbc.DropdownMenuItem(
                    children=[
                        dbc.Button(children="HTML", id=f"{self.id}-html", className="download-button"),
                    ]
                ),
                dcc.Download(id="download-file"),
                dbc.Tooltip(
                        "Download this plot to your device as a plotly JSON or interactive HTML file "
                        "for easy sharing or future use.",
                        target="dropdown-menu-button",
                    ),
            ],
            toggleClassName="dropdown-menu-toggle-class",
        )

        return dropdown_menu
```